reframe/core/config.py

```diff
@@ -440,6 +439,31 @@ def select_subconfig(self, system_fullname=None,
         self._subconfigs[system_fullname] = local_config
 
 
+def find_config_files(config_path=None, config_file=None):
+    res = []
+    if config_path:
+        for p in config_path:
+            if os.path.exists(p + '/settings.py'):
+                res.append(p + '/settings.py')
```

Comment:
Actually, why don't you construct the path once at the beginning of the for loop?